src/cc2olx/content_processors/assignment.py

```diff
@@ -0,0 +1,223 @@
+
+        return {
+            "prompt": getattr(prompt_element, "text", ""),
+            "prompts_type": self._define_prompts_type(prompt_element),
+        }
+
+    @staticmethod
+    def _define_prompt_element(resource_root: cc_xml.AssignmentElement) -> cc_xml.CommonCartridgeElementBase:
+        """
+        Define the element containing prompt data.
+        """
+        text = resource_root.get_text()
+        return text if text is not None else resource_root.instructor_text
+
+    @staticmethod
+    def _define_prompts_type(prompt_element: cc_xml.CommonCartridgeElementBase) -> str:
+        """
+        Define prompts type.
+        """
+        raw_prompt_type = prompt_element.attrib["texttype"] if prompt_element is not None else "text/plain"
+        return "html" if raw_prompt_type == "text/html" else "text"
+
+    def _parse_response_data(self, resource_root: cc_xml.AssignmentElement) -> dict:
+        """
+        Parse response-related data.
+        """
+        accepted_format_types = self._parse_accepted_format_types(resource_root)
+        is_file_submission_allowed = self._is_file_submission_allowed(accepted_format_types)
+        is_textual_submission_allowed = self._is_textual_submission_allowed(accepted_format_types)
+        file_upload_response = self._get_file_upload_response(is_textual_submission_allowed, is_file_submission_allowed)
+
+        return {
+            "text_response": self._get_text_response(is_textual_submission_allowed, is_file_submission_allowed),
+            "text_response_editor": self._get_text_response_editor(accepted_format_types),
+            "file_upload_response": file_upload_response,
+            "allow_multiple_files": True,
+            "file_upload_type": self._get_file_upload_type(file_upload_response),
+            "white_listed_file_types": self._get_white_listed_file_types(file_upload_response),
+        }
+
+    def _parse_accepted_format_types(self, resource_root: cc_xml.AssignmentElement) -> Set[str]:
+        """
+        Parse accepted format types.
+        """
+        accepted_format_types = {accepted_format.attrib["type"] for accepted_format in resource_root.accepted_formats}
+        return accepted_format_types or self.DEFAULT_ACCEPTED_FORMAT_TYPES
+
+    @staticmethod
+    def _is_file_submission_allowed(accepted_format_types: Set[str]) -> bool:
+        """
+        Decide whether submitting a file as an answer to assignment is allowed.
+        """
+        return not accepted_format_types or AssignmentSubmissionFormatType.FILE in accepted_format_types
+
+    @staticmethod
+    def _is_textual_submission_allowed(accepted_format_types: Set[str]) -> bool:
+        """
+        Decide whether submitting a textual answer to assignment is allowed.
+        """
+        return not accepted_format_types or bool(
+            AssignmentSubmissionFormatType.get_not_file_types().intersection(accepted_format_types)
```

Comment:
Nit: The name "get_not_file_types" is unclear. Since this is the only place this method is called in the code, please just do the comparison here instead, e.g.:

```python
bool(accepted_format_type & {cls.HTML, cls.TEXT, cls.URL})
```